### PR TITLE
*: golint: add and fix up comments on exported types

### PR DESCRIPTION
**Pull request by mdlayher**
Added comments and rephrased a few others.

Unexported the `extensionPair` type, because I couldn't find any place where it was returned in the exposed API.

r: @davecheney

### CODE REVIEW

File: packet.go [hunk 1]
```diff
@@ -207,7 +207,7 @@ func (p *sshFxInitPacket) UnmarshalBinary(b []byte) (err error) {
 		return err
 	}
 	for len(b) > 0 {
-		ep := ExtensionPair{}
+		ep := extensionPair{}
```
Comment (davecheney) [inline packet.go]: var ep extensionPair

File: packet.go [hunk 2]
```diff
@@ -207,7 +207,7 @@ func (p *sshFxInitPacket) UnmarshalBinary(b []byte) (err error) {
 		return err
```
Comment (davecheney) [inline packet.go]: Please remove the named return value

Comment (mdlayher) [inline packet.go]: Done.

Comment (mdlayher) [inline packet.go]: I personally don't like named returns or naked returns at all, outside of for documentation purposes in `interface` declarations.

I'd be happy to remove them throughout in a different PR, if you agree.

Comment (davecheney) [inline packet.go]: yes, please kill them with fire.

Comment (mdlayher) [inline packet.go]: Opened #76 for further discussion.

**Comment by mdlayher** [comment]
Any more thoughts before this one goes in?

**Comment by davecheney** [comment]
LGTM.

**Pull request merged**